ui/components/app/signature-request-original/signature-request-original.component.js

```diff
@@ -258,13 +241,60 @@ export default class SignatureRequestOriginal extends Component {
     );
   };
 
-  renderFooter = () => {
+  renderWarning = () => {
     const {
       cancel,
       clearConfirmTransaction,
       history,
       mostRecentOverviewPage,
       sign,
+    } = this.props;
+    const { fromAccount } = this.state;
+    const { trackEvent, type } = this.context;
+
```

Comment:
Inline callbacks on `SignatureRequestOriginalWarning` are huge, it will be better to extract them out into functions.

Reply:
Also, I would prefer a small separate component over function that returns JSX.

Reply:
Done, unit test also added @jpuri